File: newMyDataStructureAlgorithm/datastructure/hash/lib/위장.java
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
		HashMap<String, Integer> hm = new HashMap<>();
		for (int i = 0; i < clothes.length; i++) {
			if (hm.containsKey(clothes[i][1])) {
				hm.replace(clothes[i][1], hm.get(clothes[i][1]) + 1);
			} else {
				hm.put(clothes[i][1], 1);
			}
		}
		int answer = 1;
		for (int value : hm.values()) {
			answer *= (value + 1);
		}
		return --answer;
	}
}

File: programmers/level2/위장.js
function solution(clothes) {
  let answer = 1;
  let object = {};
  
  for (let i = 0; i < clothes.length; i++) {
    object[clothes[i][1]] = (object[clothes[i][1]] || 1) + 1;
  }

  for (let key in object) {
    answer *= object[key];
  }
  
  return answer - 1;
}